source_modelling/srf.py

```diff
@@ -0,0 +1,407 @@
+    try:
+        return int(int_str)
+    except ValueError:
+        if label:
+            raise SrfParseError(f'Expecting int ({label}), got: "{int_str}"')
+        else:
+            raise SrfParseError(f'Expecting int, got: "{int_str}"')
+
+
+def read_points_count(srf_file: TextIO) -> int:
+    """Read the number of points contained in an srf file.
+
+    Parameters
+    ----------
+    srf_file : TextIO
+        The SRF file object to read from.
+
+    Raises
+    ------
+    SrfParseError
+        If the current line does not match the expected syntax for the number
+        of points in an SRF file (see POINT_COUNT_RE).
+
+    Returns
+    -------
+    int
+        The number of points contained in the SRF file.
+    """
+    points_count_line = srf_file.readline().strip()
+    points_count_match = re.match(POINT_COUNT_RE, points_count_line)
+    if not points_count_match:
+        raise SrfParseError(f'Expecting POINTS header line, got: "{points_count_line}"')
+    return int(points_count_match.group(1))
+
+
+def read_srf_point(srf_file: TextIO) -> dict[str, int | float]:
+    """Read a single SRF point from a file handle.
+
+    Parameters
+    ----------
+    srf_file : TextIO
+        The file handle of the SRF file, pointing to the start of the
+        line.
+
+    Returns
+    -------
+    dict[str, int | float]
+        A dictionary containing the values for the point.
+    """
+    # skip optional first spaces
+    row = {
+        "lon": read_float(srf_file, label="lon"),
+        "lat": read_float(srf_file, label="lat"),
+        "dep": read_float(srf_file, label="dep"),
+        "stk": read_float(srf_file, label="stk"),
+        "dip": read_float(srf_file, label="dip"),
+        "area": read_float(srf_file, label="area"),
+        "tinit": read_float(srf_file, label="tinit"),
+        "dt": read_float(srf_file, label="dt"),
+        "rake": read_float(srf_file, label="rake"),
+        "slip1": read_float(srf_file, label="slip1"),
+    }
+    nt1 = read_int(srf_file, label="nt1")
+    row["slip2"] = read_float(srf_file, label="slip2")
+    nt2 = read_int(srf_file, label="nt2")
+    row["slip3"] = read_float(srf_file, label="slip3")
+    nt3 = read_int(srf_file, label="nt3")
+    row["slipt1"] = np.fromiter(
+        (read_float(srf_file, label="slipt1") for _ in range(nt1)), float
+    )
+    row["slipt2"] = np.fromiter(
+        (read_float(srf_file, label="slipt2") for _ in range(nt2)), float
+    )
+    row["slipt3"] = np.fromiter(
+        (read_float(srf_file, label="slipt2") for _ in range(nt3)), float
+    )
+    return row
+
+
+def read_srf(srf_ffp: Path) -> SrfFile:
+    """Read an SRF file into an SrfFile object.
+
+    Parameters
+    ----------
+    srf_ffp : Path
+        The filepath of the SRF file.
+
+    Returns
+    -------
+    SrfFile
+        The filepath of the SRF file.
+    """
+    with open(srf_ffp, mode="r", encoding="utf-8") as srf_file_handle:
+        version = read_version(srf_file_handle)
```

Comment:
Any particular reason for making functions for `read_version` and `read_points_count` ? They are both very short, very specific (one specific task that is generally not done by itself and requires file to be at the correct location) and only used once. Is there any scenarios where a user wouldn't use `read_srf`?